crates/fuel-core/src/service.rs

```diff
@@ -253,13 +253,20 @@ impl RunnableService for Task {
         _: &StateWatcher,
         _: Self::TaskParams,
     ) -> anyhow::Result<Self::Task> {
-        let view = self.shared.database.on_chain().latest_view();
+        let on_view = self.shared.database.on_chain().latest_view();
+        let off_view = self.shared.database.off_chain().latest_view();
         // check if chain is initialized
-        if let Err(err) = view.get_genesis() {
+        if let Err(err) = on_view.get_genesis() {
             if err.is_not_found() {
-                let result = execute_genesis_block(&self.shared.config, &view)?;
+                let result = execute_genesis_block(&self.shared.config, &on_view)?;
 
                 self.shared.block_importer.commit_result(result).await?;
+
+                let off_chain_db_transaction = genesis::off_chain::execute_genesis_block(
+                    &self.shared.config,
+                    &off_view,
+                )?;
+                off_chain_db_transaction.commit()?;
```

Comment:
Shouldn't we run both both offchain and onchain parts before committing either? Not sure if that actually matters.

Reply:
Each of them lives inside its own database, so we can't create one database transaction for both databases.